src/main/scala/com/fulcrumgenomics/commons/io/CommandLineTool.scala

```diff
@@ -0,0 +1,162 @@
+    } catch {case e: Exception => throw new Exception(s"Cannot execute script from $script with args: $args") }
+
+  /** Executes a script from the classpath. */
+  def exec(scriptResource: String, args: String*): Unit =
+    Try { writeResourceToTempFile(scriptResource) }.map(path => exec(path, args:_*))
+
+  /** Executes from a script stored at a Path. */
+  def exec(script: Path, args: String*): Unit = try {
+    val command = Executable +: script.toAbsolutePath.toString +: args
+    val process = new ProcessBuilder(command:_*).redirectErrorStream(false).start()
+    val pipe1   = Io.pipeStream(process.getErrorStream, logger.info)
+    val pipe2   = Io.pipeStream(process.getInputStream, logger.debug)
+    val retval  = process.waitFor()
+    pipe1.close()
+    pipe2.close()
+
+    if (retval != 0) throw ToolException(retval)
+  } catch {case e: Exception => throw new Exception(s"Cannot execute script from path $script with args: $args") }
+
+  /** Extracts a resource from the classpath and writes it to a temp file on disk. */
+  private def writeResourceToTempFile(resource: String): Path = {
+    val lines = Io.readLinesFromResource(resource).toSeq
+    val path  = Io.makeTempFile("script.", suffix = Suffix)
+    path.toFile.deleteOnExit()
+    Io.writeLines(path, lines)
+    path
+  }
+}
+
+trait Versioned {
+  self: CommandLineTool =>
+  val VersionFlag: String      = "--version"
+  val TestCommand: Seq[String] = Seq(VersionFlag)
+
+  /** Returns version of the tool */
+  lazy val Version: String = execArgs(Executable +: TestCommand:_*)._2
+}
+
+
+trait Modular {
+  self: CommandLineTool =>
+
+  /** Command used to test if a module is included with the tested executable. */
+  def TestModuleCommand(module: String): Seq[String]
+
+  /** Command used to test if multiple modules are included with the tested executable. */
+  def TestModuleCommand(modules: Seq[String]): Seq[Seq[String]] = modules.map(TestModuleCommand)
+
+  /** Returns true if the tested module exist with the tested executable. */
+  def IsModuleAvailable(module: String): Boolean = execArgs(TestModuleCommand(module): _*)._1
+
+  /** Returns true if all tested modules exist with the tested executable. */
+  def IsModuleAvailable(modules: Seq[String]): Boolean =  modules.map(IsModuleAvailable).forall(_ == true)
+}
+
+object Rscript extends CommandLineTool with Versioned with Modular with CanRunScript {
+  val Executable: String = "Rscript"
+  val Suffix: String     = ".R"
+  def TestModuleCommand(module: String): Seq[String] = Seq(Executable, "-e", s"stopifnot(require('$module'))")
+
+  /** Only returns true if R exists and ggplot2 is installed */
+  override lazy val Available: Boolean = {
+    val ToolAvailable: Boolean    = execArgs(Executable +: Seq(VersionFlag):_*)._1
+    val ModuleAvailable : Boolean = IsModuleAvailable(module = "ggplot2")
+    Seq(ToolAvailable, ModuleAvailable).forall( _ == true)
+  }
+}
+
+object GhostScript extends CommandLineTool with Versioned {
```

Comment:
I would say leave this out of this PR and only include vendored support for Python and Rscript.